Cython/Compiler/FlowControl.py

```diff
@@ -1069,6 +1069,14 @@ def visit_ParallelWithBlockNode(self, node):
 
         return node
 
+    def visit_DeviceWithBlockNode(self, node):
```

Comment:
1. Have you missed a `return node`?
2. Given that this is (otherwise) the same as `visit_ParallelWithBlockNode` could you avoid duplication by calling that (or possibly inheritance)

Reply:
You're right. I will make the changes as suggested.